webpack/assets/javascripts/react_app/components/factCharts/index.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Button, Modal } from 'react-bootstrap';
+import helpers from '../../common/helpers';
+import PieChart from '../common/charts/PieChart';
+import Loader from '../common/Loader';
+import MessageBox from '../common/MessageBox';
+import { STATUS } from '../../constants';
+import * as FactChartActions from '../../redux/actions/factCharts/';
+
+class FactChart extends React.Component {
+  constructor(props) {
+    super(props);
+    helpers.bindMethods(this, ['onClick', 'closeModal']);
+  }
+
+  onClick() {
+    const { showModal, getChartData, data: { id, path, title } } = this.props;
+
+    getChartData(path, id);
+    showModal(id, title);
+  }
+
+  closeModal() {
+    this.props.closeModal();
+  }
+
+  render() {
+    const { factChart, data: { id, title }, modalToDisplay } = this.props;
+
+    const chart = modalToDisplay ? (<PieChart
+        data={factChart.chartData}
+        key={`chart-${id}`}
+        smallChart = {false}
+      />) : false;
+
+    const requestErrorMsg = factChart.loaderStatus === STATUS.ERROR ? __('Request Failed') : undefined;
```

Comment:
what do you think about moving `__('No data available')` to here instead of undefined?